Sources/Common/Communication/EventBusHandler.swift

```diff
@@ -1,66 +1,87 @@
 import Combine
 import Foundation
 
+/// Manages the handling of events through an event bus and coordinates with event storage.
+/// It handles both in-memory and file-based storage of events.
+// sourcery: InjectRegisterShared = "EventBusHandler"
+// sourcery: InjectSingleton
 public class EventBusHandler {
     private let eventBus: EventBus
     private let eventStorage: EventStorage
+    private var memoryStorage: [String: [any EventRepresentable]] = [:]
 
+    /// Initializes the EventBusHandler with an EventBus and EventStorage.
+    /// Automatically loads events from file-based storage into in-memory storage upon initialization.
     public init(eventBus: EventBus, eventStorage: EventStorage) {
         self.eventBus = eventBus
         self.eventStorage = eventStorage
+        loadEventsFromStorage()
```

Comment:
I wonder if calling this function directly in the constructor could cause lower performance of the SDK, if the constructor is called on the main thread? 

Perhaps, the `load` function is an async function that executes on a background thread? Or, it's a synchronous function but we add a comment suggesting it gets called from a background thread?

Reply:
The implementation in this PR is more simple. I prefer to not need to involve threading if we can help it. I agree with you that I also do not anticipate issues at this time. 

I think it would be good to add a comment reminding us of this decision. Here is a suggestion to try and help you out. 

```suggestion
        // We do not anticipate the number of events in storage to be a large number at this time. Therefore, we have decided to use this more simple approach of loading events in a synchronous way and not enforcing it be done on a different thread. 
        loadEventsFromStorage()
```